lib/puppet/provider/service/init.rb

```diff
@@ -152,6 +152,15 @@ def restartcmd
     (@resource[:hasrestart] == :true) && [initscript, :restart]
   end
 
+  def texecute(type, command, fof = true, squelch = false, combine = true)
+    if type =~ /^start/
+      if Facter.value(:operatingsystem) == "Solaris"
+        command =  ["/usr/bin/ctrun -l none", command].flatten.join(" ")
```

Comment:
I think this is good. We know we're calling flatten on a non-nil Array, and flatten produces a single array with at least one element (since we hardcoded `/usr/bin/ctrun ...`):

```
irb(main):009:0> [].join(' ')
=> ""
irb(main):010:0> ['foo'].join(' ')
=> "foo"
irb(main):011:0> ['foo', 'bar'].join(' ')
=> "foo bar"
```

I think the possible bad input would be if `command` included something that wasn't a String or Array, e.g.

```
irb(main):013:0> ['foo', { 1 => 2 } ].join(' ')
=> "foo {1=>2}"
```

But I'm pretty sure that wouldn't happen (since `command` gets passed to `Puppet::Util::Execution.execute` and that at least documents it as an array of strings https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/execution.rb#L112-L114